### ISSUE TITLE
Footstep planning.

### ISSUE THREAD

**Issue by jeljaik** [jeljaik/WB-Team-Issues#15]
- [x] Describe plan for next two weeks in this issue.

**Comment by jeljaik** [comment]
cc @luca-fiorio @francesco-romano

**Comment by jeljaik** [comment]
**Details on Joint Limits**
Right leg
- (5) ankle_roll_r = [-25.4  26.6](-1deg when hitting ankle_pitch_r limits)
- (4) ankle_pitch_r = [-39.6  42.0]
- (3) knee_r = [-114.3  3.5]
- (2) hip_yaw_r  = [as they were] 
- (1) hip_roll_r = [-22.3  84.0]
- (0) hip_pitch_r = [-36.4  80.6]

Left leg
- (5) ankle_roll_l = [-26.2  25.6]
- (4) ankle_pitch_l = [-37.6  40.7]
- (3) knee_l = [-115.6  4.5]
- (2) hip_yaw_l = [as they were]
- (1) hip_roll_r = [-20.3  76.9]
- (0) hip_pitch_r = [-36.2  79.8]

Torso 
- (0) torso_yaw   = [-64.1  58.9]
- (1) torso_roll    = [-30.5 30.5]
- (2) torso_pitch = [-21.8  70]

**Comment by fiorisi** [comment]
From Black iCub:
DEGREES 
leg
- (5) ankle_roll = [-25 25]
- (4) ankle_pitch = [-35 40]
- (3) knee = [-114 3]
- (2) hip_yaw = [as they were]
- (1) hip_roll = [-20 75]
- (0) hip_pitch = [-35 78]

Torso
- (0) torso_yaw = [-64 58]
- (1) torso_roll = [-30 30]
- (2) torso_pitch = [-20 69]

RADIANTS
leg
- (5) ankle_roll = [-0.4363 0.4363]
- (4) ankle_pitch = [-0.6109 0.6981]
- (3) knee = [-1.9897 0.0524]
- (2) hip_yaw = [as they were]
- (1) hip_roll = [-0.3491 1.3090]
- (0) hip_pitch = [-0.6109 1.3614]

Torso
- (0) torso_yaw = [-1.1170 1.0123]
- (1) torso_roll = [-0.5236 0.5236]
- (2) torso_pitch = [-0.3491 1.2043]

**Comment by jeljaik** [comment]
As of today, iCub is able to walk in a new open loop fashion with the capture point approach in Simulation!!! thanks to the trajectories generated by @luca-fiorio and the help of @jeljaik and @francesco-romano. This post serves as a log entry for future references to replicate our results. We have experienced in the past that things work on one computer only (right @DanielePucci?) and this is not desirable at all. I will try to be as detailed as possible. In case I'm missing something you may consider relevant, ask away!

**PC Specifications**
- **Lenovo X220** Memory: 7,7 GiB, Intel Core ® i5-2520M CPU @ 2.5 GHZ. Graphics: Intel ® Sandybridge Mobile. OS Type: 64 bit. OS: Ubuntu 14.04 LTS. 

**Module Specifications**
- For playing back the trajectories the `walkPlayer` module was used and it can be found here: https://github.com/robotology/codyco-modules/tree/new_wbi_ID/src/modules/walkPlayer
- Update `codyco-modules` in the branch `new_wbi_ID` and install the `walkPlayer`.
- To launch it do: `walkPlayer --robot icubGazeboSim --filename2 softTouchDown --execute`
- To play, in another terminal do: `yarp rpc /walkPlayer/rpc` followed by `start`. `stop` is another accepted command.
- Version of the `walkPlayer`: 
  https://github.com/robotology/codyco-modules/commit/e788c8573230b8afa1ee559f9959766eb509dd05

**Dependencies and miscellaneous Specifications**
- Gazebo Version: 4.1.0
- yarp version: https://github.com/robotology/yarp/commit/317184603fe38d89b84a0b6162cac979dcca69f7
- iCub version: https://github.com/robotology/icub-main/commit/aec92a3326b3d6a29af95864ae7bceade425c193
- gazebo-yarp-plugins version: https://github.com/robotology/gazebo-yarp-plugins/commit/f92a8f2772e3931a9686a573e3947ae3cb082ca9
- Sequences used: https://github.com/robotology/codyco-modules/commit/8bd75f433a3cf79b4338eddaf11679adf14dd9b2
- icub-gazebo version (`walking` branch): https://github.com/robotology-playground/icub-gazebo/commit/480ad3897afb9e9b819fdb963943fb61160afe3d

**Notes** Bear in mind that this walking cannot be used with the real robot, because the real joint limits, in particular the feet pitch lower limits have been increased and we cannot reach them with the real iCub. A new trajectory that respects these limits is the current work in progress.

**Comment by traversaro** [comment]
The iCub sdf is the one in
https://github.com/robotology-playground/icub-gazebo ?

**Comment by jeljaik** [comment]
Not exactly, I just created a branch called `walking`. In this branch I've just changed the lower joint limit of the feet pitch to -0.8 and the PID gains have been increased 10 times except for the knees.  
https://github.com/robotology-playground/icub-gazebo/commit/480ad3897afb9e9b819fdb963943fb61160afe3d

**Comment by jeljaik** [comment]
However, the joint limits written above have been updated in https://github.com/robotology-playground/icub-gazebo/commit/ea66269a9d80f289d764076189608d897a8d847d

**Comment by jeljaik** [comment]
Pending Task:
- [ ] Generate H_dot_des for the future torque-based walking
